pkg/api/handlers/memorysets.go

```diff
@@ -0,0 +1,168 @@
+	for i := range memories {
+		memory := memories[i]
+		// Generate ID from content hash if not provided
+		if memory.ID == "" {
+			// Create a shorter URL-friendly hash (first 12 chars of SHA-256)
+			fullHash := hash.String(memory.Content)
+			memory.ID = fullHash[:12]
+		}
+		// Always set creation time to current time
+		memory.CreatedAt = *currentTime
+		newMemories[memory.ID] = memory
+	}
+
+	if err := req.Get(&memorySet, thread.Name); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		// Create a new MemorySet
+		memorySet = v1.MemorySet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      thread.Name,
+				Namespace: req.Namespace(),
+			},
+			Spec: v1.MemorySetSpec{
+				ThreadName: thread.Name,
+				Manifest: types.MemorySetManifest{
+					Memories: []types.Memory{},
+				},
+			},
+		}
+
+		for _, memory := range newMemories {
+			memorySet.Spec.Manifest.Memories = append(memorySet.Spec.Manifest.Memories, memory)
+		}
+
+		if err := req.Create(&memorySet); err != nil {
+			return err
+		}
+	} else {
+		// Add memories to the existing MemorySet
+		existingMap := make(map[string]struct{})
+		for _, memory := range memorySet.Spec.Manifest.Memories {
+			existingMap[memory.ID] = struct{}{}
+		}
+
+		for id, memory := range newMemories {
+			if _, exists := existingMap[id]; exists {
+				continue
+			}
+
+			memorySet.Spec.Manifest.Memories = append(memorySet.Spec.Manifest.Memories, memory)
+		}
+
+		if err := req.Update(&memorySet); err != nil {
+			return err
+		}
+	}
+
+	return req.Write(&types.MemorySet{
+		Metadata: types.Metadata{
+			ID: memorySet.Name,
+		},
+		MemorySetManifest: memorySet.Spec.Manifest,
+	})
+}
+
+func (*MemorySetHandler) GetMemories(req api.Context) error {
+	var memorySet v1.MemorySet
+	thread, err := getThreadForScope(req)
+	if err != nil {
+		return err
+	}
+
+	if err := req.Get(&memorySet, thread.Name); err != nil {
+		return err
+	}
+
+	return req.Write(&types.MemorySet{
+		Metadata: types.Metadata{
+			ID: memorySet.Name,
+		},
+		MemorySetManifest: memorySet.Spec.Manifest,
+	})
+}
+
+func (*MemorySetHandler) DeleteMemories(req api.Context) error {
+	thread, err := getThreadForScope(req)
+	if err != nil {
+		return err
+	}
+
+	// Check if a specific memory_id is provided
+	memoryID := req.PathValue("memory_id")
+	if memoryID == "" {
+		// No memory_id provided, delete the entire memory set
+		return req.Delete(&v1.MemorySet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      thread.Name,
+				Namespace: req.Namespace(),
+			},
+		})
+	}
+
+	// Memory ID provided, delete just that specific memory
+	var memorySet v1.MemorySet
+	if err := req.Get(&memorySet, thread.Name); err != nil {
+		return err
+	}
+
+	// Create a new slice without the memory to delete
+	var updatedMemories []types.Memory
+	for _, memory := range memorySet.Spec.Manifest.Memories {
+		if memory.ID == memoryID {
+			continue
+		}
+		updatedMemories = append(updatedMemories, memory)
+	}
```

Comment:
Do we need to create an entire new slice? Seems like something like `slices.DeleteFunc` would work here.

If we do need to create a new slice, then:
```suggestion
	// Create a new slice without the memory to delete
	updatedMemories := make([]types.Memory, 0, len(memorySet.Spec.Manifest.Memories))
	for _, memory := range memorySet.Spec.Manifest.Memories {
		if memory.ID == memoryID {
			continue
		}
		updatedMemories = append(updatedMemories, memory)
	}
```